src/momento/internal/aio/_scs_data_client.py

```diff
@@ -181,6 +212,166 @@ async def execute_delete_request_fn(req: _DeleteRequest) -> _DeleteResponse:
         )
 
     # DICTIONARY COLLECTION METHODS
+    async def dictionary_get_fields(
+        self,
+        cache_name: TCacheName,
+        dictionary_name: TDictionaryName,
+        fields: TDictionaryFields,
+    ) -> CacheDictionaryGetFieldsResponse:
+        try:
+            self._log_issuing_request("DictionaryGet", {"dictionary_name": dictionary_name})
+            _validate_cache_name(cache_name)
+            _validate_dictionary_name(dictionary_name)
+
+            request = _DictionaryGetRequest()
+            request.dictionary_name = _as_bytes(dictionary_name, self.__UNSUPPORTED_DICTIONARY_NAME_TYPE_MSG)
+            bytes_fields = cast(
+                List[bytes], _dictionary_fields_as_bytes(fields, self.__UNSUPPORTED_DICTIONARY_FIELDS_TYPE_MSG)
+            )
+            request.fields.extend(bytes_fields)
+
+            response = await self._build_stub().DictionaryGet(
+                request,
+                metadata=make_metadata(cache_name),
+                timeout=self._default_deadline_seconds,
+            )
+            self._log_received_response("DictionaryGet", {"dictionary_name": dictionary_name})
+
+            type = response.WhichOneof("dictionary")
+            if type == "found":
+                get_responses: List[CacheDictionaryGetFieldResponse] = []
+                for field, get_response in zip(bytes_fields, response.found.items):
+                    if get_response.result == Miss:
+                        get_responses.append(CacheDictionaryGetField.Miss())
+                    else:
+                        get_responses.append(CacheDictionaryGetField.Hit(get_response.cache_body, field))
+                return CacheDictionaryGetFields.Hit(get_responses)
+            elif type == "missing":
+                return CacheDictionaryGetFields.Miss()
+            else:
+                raise UnknownException("Unknown dictionary field")
+        except Exception as e:
+            self._log_request_error("dictionary_get_fields", e)
+            return CacheDictionaryGetFields.Error(convert_error(e))
+
+    async def dictionary_fetch(
+        self, cache_name: TCacheName, dictionary_name: TDictionaryName
```

Comment:
same questions I asked on @schwern 's PR:

1. Is the type definition file you have created valuable enough to justify the fact that future developers cannot look at this file in isolation and know what the types are?  Could we not localize some of the type aliases?
2. What shows up to users when they get hints in their IDE?  Is it TCacheName (not useful)?  Or `str` (useful)?

Reply:
> same questions I asked on @schwern 's PR:
> 
> 1. Is the type definition file you have created valuable enough to justify the fact that future developers cannot look at this file in isolation and know what the types are?  Could we not localize some of the type aliases?
The union types are definitely very busy to keep repeating everywhere. 

> 2. What shows up to users when they get hints in their IDE?  Is it TCacheName (not useful)?  Or `str` (useful)?
Hovering over the type hint resolves to the concrete types

Reply:
1. how many files would they actually be repeated in though?  And would it be really bad to just have a local type alias in each file that needed one?

2. What I'm asking is if I'm a user and I type `dictionary_fetch(` and then get my IDE to pop up the hints for the parameter list, is it going to show me `TCacheName`?  Or the real type?

Reply:
1. A lot. We've already changed the types several times, so local type aliases wouldn't work. They've proven very valuable.
2. VSCode...
<img width="640" alt="Screen Shot 2023-02-03 at 15 59 38" src="https://user-images.githubusercontent.com/25888/216732553-9ae5707d-03cd-42a1-b76c-5050a079df4f.png">

Reply:
thanks.  if it shows up like that in VSCode and IntelliJ then I can live with it.

Reply:
PyCharm is good with it.

<img width="975" alt="Screen Shot 2023-02-03 at 17 00 48" src="https://user-images.githubusercontent.com/25888/216736711-2d7dba00-07f6-4f25-a1fd-53f32cfa5f55.png">

Reply:
That's on the function definition itself, but if you write out in a separate page of code, `client.dictionary_set_fields(`, then you get the type aliases in the tooltip, and they do not resolve to the original types. I've been poking around to see if there is a way to get both but I cannot find one.

Reply:
Let's make this a separate issue.

Reply:
#246